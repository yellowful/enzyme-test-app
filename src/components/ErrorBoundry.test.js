import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundry";

describe('Tests of Error Boundry',()=>{
    const wraper = shallow(<ErrorBoundary />)
    // 將class ErrorBoundry instantiate，如此等一下才可以呼叫this.setState
    const instance = wraper.instance();
    // 所有測試前先把state設成false
    beforeEach(()=>{
        instance.setState({hasError:false})
    });
    // 記錄state是false的snapshot
    it('initial state snapshot',()=>{
        expect(wraper).toMatchSnapshot();
    })
    // 記錄state是true的snapshot
    it('state test',()=>{
        instance.setState({hasError:true})
        expect(wraper).toMatchSnapshot();
    })
})