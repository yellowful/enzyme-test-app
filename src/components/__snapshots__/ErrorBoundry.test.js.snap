// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of Error Boundry initial state snapshot 1`] = `""`;

exports[`Tests of Error Boundry state test 1`] = `
<h1>
  Something went wrong.
</h1>
`;
